XLS-30/README.md

```diff
@@ -0,0 +1,1254 @@
+- `Amount` , `Amount2`, `[LPTokenOut]`
+- `Amount` and `LPTokenOut`
+- `Amount` and `EPrice`
+
+Implementation details for above combinations:
+
+1. Fields specified: `LPTokenOut`, `[Amount]`, `[Amount2]`
+
+Such a transaction assumes proportional deposit of pools assets in exchange for the specified amount of `LPTokenOut` of the AMM instance.
+
+- Use equations 1 & 2 to compute the amount of each asset to be deposited, given the amount of $\Delta_{LPTokenOut}$ specified as `LPTokenOut`
+
+If the account that initiated the transaction holds sufficient balances, the transaction is successful. It fails otherwise with an error code.
+
+Similarly, if `[Amount]`, `[Amount2]` are specified and the amount of either asset computed above is less than the amounts specified in these fields, then the transaction fails.
+
+
+2. Fields specified: `Amount`, `[LPTokenOut]`
+
+Such a transaction assumes single asset deposit of the amount of asset specified by `Amount`. This is essentially an _equal_ asset deposit and a _swap_.
+
+- Check if the account that initiated the transaction holds the amount of `Amount`. 
+- If not, the transaction fails with an error code. Otherwise,
+  - Use equation 3 to compute amount of `LPTokenOut` ~ $\Delta_{LPTokenOut}$ to be issued, given either the amount of drops of `XRP` or the amount of tokens in `Amount`.
+
+If the asset to be deposited is a token, specifying the `value` field is required, else the transaction will fail.
+
+If `LPTokenOut` field exists, and the amount of `LPTokenOut` computed above is less than the that specified in this field, then the transaction fails.
+
+3. Fields specified: `Amount`, `Amount2` and `[LPTokenOut]`
+
+Such a transaction assumes proportional deposit of pool assets with the constraints on the maximum amount of each asset that the trader is willing to deposit.
+
+- Use equation 1 to compute the amount of $\Delta_{LPTokenOut}$, given the amount in `Amount`. Let this be `Z`
+- Use equation 2 to compute the amount of asset2, given $\Delta_{LPTokenOut}$  ~ `Z`. Let the computed amount of asset2 be `X`.
+  - If `X` <= amount in `Amount2`:
+    - The amount of asset1 to be deposited is the one specified in `Amount`
+    - The amount of asset2 to be deposited is `X`
+    - The amount of `LPTokenOut` to be issued is `Z`
+  - If X > amount in `Amount2`:
+    - Use equation 2 to compute $\Delta_{LPTokenOut}$, given the amount in `Amount2`. Let this be `W`
+    - Use equation 1 to compute the amount of asset1, given $\Delta_{LPTokenOut}$ ~ `W` from above. Let the computed amount of asset1 be `Y`
+    - If `Y` <= amount in `Amount`:
+      - The amount of asset1 to be deposited is `Y`
+      - The amount of asset2 to be deposited is the one specified in `Amount2`
+      - The amount of `LPTokenOut` to be issued is `W`
+
+else, failed transaction.
+
+If `LPTokenOut` field exists, and the amount of `LPTokenOut` computed above is less than the that specified in this field, then the transaction fails.
+
```

Comment:
`Amount`, `Amount2`, [`TradingFee`] for `tfTwoAssetsIfEmpty`. This option is equivalent to `AMMCreate`, i.e. the pool is created with `Amount` and `Amount2` balances.

Reply:
Added this as 6th combination in the AMMDeposit section.